src/util/system.cpp

```diff
@@ -753,8 +753,9 @@ const fs::path &GetDataDir(bool fNetSpecific)
     if (!path.empty())
         return path;
 
-    if (gArgs.IsArgSet("-datadir")) {
-        path = fs::system_complete(gArgs.GetArg("-datadir", ""));
+    std::string datadir = gArgs.GetArg("-datadir", "");
```

Comment:
In commit "Add CheckDataDirOption() function" (5df1581b9ee6f13593b33fa9629ea9fbadcad732)

Ideally commit message might also mention that it changes the behavior of the GetDataDir function. The effect I think is to now allow `-nodatadir` or `-datadir=''` options on the command line to override any datadir specified in the config file and reset it to default. In theory this could even be mentioned in release notes, but it's probably too niche.

Reply:
> The effect I think is to now allow `-nodatadir` or `-datadir=''` options on the command line to override any datadir specified in the config file and reset it to default.

Unfortunately, neither `-nodatadir` nor `-datadir=''` on the command line can unset datadir specified in the config file or reset it to default. See: #16097.

Reply:
Then, why is this changed?

Reply:
The [initial idea](https://github.com/bitcoin/bitcoin/pull/15864#discussion_r281785729) was:
> I don't think `IsArgSet` is the right function to call here.

If we recall:
https://github.com/bitcoin/bitcoin/blob/8c69fae94410f54bad13be0f34d54370fddbf4b3/src/util/system.cpp#L470-L474

using `IsArgNegated` is (logically) inappropriate for `-datadir` option, IMO.

Also (but minor), the suggested code uses only one function call (`GetArg`) instead of two calls (`IsArgSet` and `GetArg`).

Reply:
@ryanofsky @MarcoFalke 
> The effect I think is to now allow `-nodatadir` or `-datadir=''` options on the command line to override any datadir specified in the config file and reset it to default.

Actually, `-datadir=''` (with empty string) or `-datadir` will reset it to default. So, going to close #16416 in favor of this PR.